<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="False discovery rate estimation">

<title>BINF301 Machine Learning for Systems Biology - FDR estimation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../statistical-significance/statistical-significance-notebook.html" rel="next">
<link href="../statistical-significance/statistical-significance-gws.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">BINF301 Machine Learning for Systems Biology</span>
    </a>
  </div>
        <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../statistical-significance/statistical-significance-gws.html">Statistical significance</a></li><li class="breadcrumb-item"><a href="../statistical-significance/fdr-estimation.html">Theory</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../julia-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction to Julia</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Cluster analysis</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/cancer-subtypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/combinatorial-clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/cluster-analysis-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Statistical significance</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/statistical-significance-gws.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/fdr-estimation.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/statistical-significance-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Regularized regression</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/drug-sensitivity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/ridge-lasso-elnet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/regularized-regression-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">Dimensionality reduction</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory I</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/tsne-umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory II</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">Causal inference</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-assoc-genes-disease.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/mediation-iv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-inference-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Graphical models</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/gene-regulatory-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/bayesian-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Gaussian processes</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gaussian-processes/variance-components-genomics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gaussian-processes/gaussian-processes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">Neural networks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../neural-networks/protein-structure-prediction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../neural-networks/neural-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">Appendix</span></span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#statistical-significance" id="toc-statistical-significance" class="nav-link active" data-scroll-target="#statistical-significance">Statistical significance</a></li>
  <li><a href="#false-discovery-rate" id="toc-false-discovery-rate" class="nav-link" data-scroll-target="#false-discovery-rate">False discovery rate</a>
  <ul class="collapse">
  <li><a href="#plug-in-estimator-of-the-fdr" id="toc-plug-in-estimator-of-the-fdr" class="nav-link" data-scroll-target="#plug-in-estimator-of-the-fdr">Plug-in estimator of the FDR</a></li>
  <li><a href="#bayesian-estimate-of-the-fdr" id="toc-bayesian-estimate-of-the-fdr" class="nav-link" data-scroll-target="#bayesian-estimate-of-the-fdr">Bayesian estimate of the FDR</a></li>
  <li><a href="#q-values" id="toc-q-values" class="nav-link" data-scroll-target="#q-values"><span class="math inline">\(Q\)</span>-values</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../statistical-significance/statistical-significance-gws.html">Statistical significance</a></li><li class="breadcrumb-item"><a href="../statistical-significance/fdr-estimation.html">Theory</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">FDR estimation</h1>
</div>

<div>
  <div class="description">
    False discovery rate estimation
  </div>
</div>


<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Book sections
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="https://hastie.su.domains/ElemStatLearn/"><em>Elements of Statistical Learning</em></a>: <strong>Section 18.7</strong></li>
<li><a href="https://www.statlearning.com/"><em>An Introduction to Statistical Learning</em></a>: <strong>Section 13.4</strong></li>
</ul>
</div>
</div>
<section id="statistical-significance" class="level2">
<h2 class="anchored" data-anchor-id="statistical-significance">Statistical significance</h2>
<p>When we analyze genome-wide data, we are usually interested in identifying <strong>statistically significant</strong> features or patterns, that is, patterns we would not expect to see purely by chance. For example:</p>
<ul>
<li>Genes differentially expressed between healthy and disease samples, treated and untreated samples, etc.</li>
<li>Genes coexpressed across tumour samples</li>
<li>…</li>
</ul>
<p>We will use differential expression between two conditions, treated and untreated, as a running example. Of course this applies equally to other types of comparisons, e.g.&nbsp;ER positive vs ER negative breast tumours.</p>
<p>Statistical significance is expressed in terms of comparing a <strong>null hypothesis</strong> <span class="math inline">\(H_0\)</span> to an <strong>alternative hypothesis</strong> <span class="math inline">\(H_1\)</span>. For instance</p>
<p><span class="math display">\[
\begin{aligned}
  H_0 &amp;= \text{treatment has no effect on gene expression}\\\\
  H_1 &amp;= \text{treatment does have an effect on gene expression}
\end{aligned}
\]</span></p>
<p>To compare gene expression between the two groups, we can for instance do a <a href="https://en.wikipedia.org/wiki/Student%27s_t-test">two-sample <span class="math inline">\(t\)</span>-test</a>: compute for each gene</p>
<p><span class="math display">\[
t = \frac{\overline{x_1} - \overline{x_2}}{\text{se}}
\]</span></p>
<p>where <span class="math inline">\(\overline{x_1}\)</span> and <span class="math inline">\(\overline{x_2}\)</span> are the average expression levels of the gene in each group, and <span class="math inline">\(\text{se}\)</span> is the pooled within-group standard error:</p>
<p><span class="math display">\[
\text{se} = \sqrt{\frac{s_1^2}{n_1}+\frac{s_2^2}{n_2}}
\]</span></p>
<p>where <span class="math inline">\(s_i^2\)</span> and <span class="math inline">\(n_i\)</span> are the variance and number of samples in group <span class="math inline">\(i\)</span>, respectively</p>
<p><span class="math display">\[
\begin{aligned}
  \overline{x_i} &amp;= \frac{1}{n_i}\sum_{j\in G_i} x_{j} \\\\
  s_i^2 &amp;= \frac{1}{(n_i-1)}\sum_{j\in G_i} (x_i - \overline{x_i})^2
\end{aligned}
\]</span></p>
<p>The <span class="math inline">\(t\)</span>-statistic measures the difference in average expression between the two groups, relative to the natural variation within each group. The greater <span class="math inline">\(|t|\)</span>, the more likely there is a true difference between the groups. Nevertheless, even if there is no true difference, some variation between the groups will be observed in a finite number of samples due to random sampling noise. The distribution of <span class="math inline">\(t\)</span>-values observed between two groups when in fact there is no true difference between them is called the <strong>null distribution</strong>. Statistical significance is usually expressed by the <span class="math inline">\(p\)</span>-value, which, for a given value of <span class="math inline">\(t\)</span>, expresses the probability to observe a <span class="math inline">\(t\)</span>-statistic greater than <span class="math inline">\(t\)</span> (in absolute value) under the null hypothesis:</p>
<p><span class="math display">\[
\begin{aligned}
  p = \text{Pr}(|T|\geq |t| \mid H_0)
\end{aligned}
\]</span></p>
<p>For some test statistics, the true null distribution to compute the <span class="math inline">\(p\)</span>-value is known. Otherwise random permutations can be used: randomly permute the group labels of the samples a large number of times, compute the test statistic for each permutation, and use these values to construct an <strong>empirical null distribution</strong>.</p>
<p>An important property of <span class="math inline">\(p\)</span>-values is the following: under the null hypothesis, <span class="math inline">\(p\)</span>-values are uniformly distributed between 0 and 1:</p>
<p><span class="math display">\[
\text{Pr}(P\leq p \mid H_0) = p
\]</span></p>
<p>Conventionally, a <span class="math inline">\(p\)</span>-value threshold of 0.05 is often used. This means that the probability that we declare a gene as being affected by the treatment, even though in reality it is not affected, is less than 5%.</p>
<p>In genome-wide studies, we typically measure around 20,000 genes (in humans). From the uniformity of the <span class="math inline">\(p\)</span>-value distribution under the null hypothesis, it follows that even if none of the genes are affected by treatment, we would still expect 5% or 1,000 genes to have a <span class="math inline">\(p\)</span>-value less than 0.05. Clearly it would be wrong to call any of these genes differentially expressed!</p>
</section>
<section id="false-discovery-rate" class="level2">
<h2 class="anchored" data-anchor-id="false-discovery-rate">False discovery rate</h2>
<p>The <span class="math inline">\(p\)</span>-value is often wrongly interpreted as saying that the probability that our gene is truly affected by treatment is greater than 95% if <span class="math inline">\(p&lt;0.05\)</span>. The latter is the probability that the alternative hypothesis is true given that the <span class="math inline">\(p\)</span>-value is below a certain threshold, or <span class="math inline">\(\text{Pr}(H_1 \mid P\leq p)\)</span>. More commonly, we express this through the <strong>false discrovery rate (FDR)</strong>, that is, the probability that the null hypothesis is true given that the <span class="math inline">\(p\)</span>-value is below a certain threshold, <span class="math inline">\(\text{Pr}(H_0 \mid P\leq p)=1-\text{Pr}(H_1 \mid P\leq p)\)</span>.</p>
<p>To get a feel for what such probabilities mean, consider the following table:</p>
<table class="table">
<thead>
<tr class="header">
<th></th>
<th>Significant</th>
<th>Not Significant</th>
<th>Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong><span class="math inline">\(H_0\)</span> true</strong></td>
<td><span class="math inline">\(F\)</span></td>
<td><span class="math inline">\(M_0-F\)</span></td>
<td><span class="math inline">\(M_0\)</span></td>
</tr>
<tr class="even">
<td><strong><span class="math inline">\(H_1\)</span> true</strong></td>
<td><span class="math inline">\(T\)</span></td>
<td><span class="math inline">\(M_1-T\)</span></td>
<td><span class="math inline">\(M_1\)</span></td>
</tr>
<tr class="odd">
<td><strong>Total</strong></td>
<td><span class="math inline">\(S\)</span></td>
<td><span class="math inline">\(M-S\)</span></td>
<td><span class="math inline">\(M\)</span></td>
</tr>
</tbody>
</table>
<p>Here “F” stands for “false positive” (not affected by treatment, but still called significant), “T” for “true positive”. The second column contains respectively the “true negative” and “false negative” results.</p>
<p>Then the false discovery rate is</p>
<p><span class="math display">\[
\text{FDR} = \mathbb{E}\left(\frac{F}{S}\right)
\]</span></p>
<p>Obviously, we never know the true value of the numbers in the table, and hence FDR must be estimated. There exist several approaches for doing this. We consider two popular ones: the plug-in estimator and a Bayesian approach.</p>
<section id="plug-in-estimator-of-the-fdr" class="level3">
<h3 class="anchored" data-anchor-id="plug-in-estimator-of-the-fdr">Plug-in estimator of the FDR</h3>
<p>Consider again the differential expression problem. Compute <span class="math inline">\(t\)</span>-statistics for each gene and consider a range of thresholds <span class="math inline">\(C\)</span>, either on the absolute <span class="math inline">\(t\)</span>-values or on the corresponding <span class="math inline">\(p\)</span>-values. For each value of <span class="math inline">\(C\)</span>, we know <span class="math inline">\(S_{obs}(C)\)</span>, the observed number of significant genes at threshold <span class="math inline">\(C\)</span>.</p>
<p>Now generate <span class="math inline">\(K\)</span> random permutations of the group labels, and compute for each permutation <span class="math inline">\(k\)</span>, <span class="math inline">\(F_{obs}(C,k)\)</span>, the number of significant genes at threshold <span class="math inline">\(C\)</span> in permutation <span class="math inline">\(k\)</span>, which by virtue of the randomization, must all correspond to cases where <span class="math inline">\(H_0\)</span> is true. Compute the average over all permutations:</p>
<p><span class="math display">\[
F_{obs}(C) = \frac{1}{K}\sum_{k=1}^K F_{obs}(C,k)
\]</span></p>
<p>The plug-in estimate of the FDR at threshold <span class="math inline">\(C\)</span> is then defined as</p>
<p><span class="math display">\[
\widehat{\text{FDR}}(C) = \frac{F_{obs}(C)}{S_{obs}(C)}
\]</span></p>
<p>We can now vary <span class="math inline">\(C\)</span> until we reached a desired FDR value, e.g.&nbsp;10%, such that we expect no more than 10% of the genes we call differentially expressed to be false positives.</p>
<p>Note that in practice, the number of permutations <span class="math inline">\(K\)</span> need not be very large to reach a stable value for the average. This is because in each permutation, we obtain random <span class="math inline">\(p\)</span>-values for a large number of features (genes).</p>
</section>
<section id="bayesian-estimate-of-the-fdr" class="level3">
<h3 class="anchored" data-anchor-id="bayesian-estimate-of-the-fdr">Bayesian estimate of the FDR</h3>
<p>When we perform a genome-wide test for differential expression, we obtain on the order of <span class="math inline">\(10^4\)</span> test statistic values, with corresponding <span class="math inline">\(p\)</span>-values under the null hypothesis. If all is well, the distribution of these <span class="math inline">\(p\)</span>-values, visualized using a histogram, typically looks like the <em>anti-conservative p-values</em> on this page:</p>
<p><a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/">How to interpret a <span class="math inline">\(p\)</span>-value histogram</a></p>
<p>We know that under the null hypothesis, <span class="math inline">\(p\)</span>-values are uniformly distributed. We recognize the uniform distribution in the flat profile of the histogram for unsignificant (<span class="math inline">\(p\to 1\)</span>) <span class="math inline">\(p\)</span>-values, which with very high probability come from genes unaffected by the treatment. At very small <span class="math inline">\(p\)</span>-values, we see a deviation from the uniform distribution, which indicates the presence of truly affected genes, which obsviously don’t follow the null distribution.</p>
<p>Hence it seems like we can model the observed <span class="math inline">\(p\)</span>-value distribution as a <strong>mixture distribution</strong> with one component corresponding to the null distribution and one component corresponding to the alternative distribution:</p>
<p><span class="math display">\[
f(p) = \pi_0 f_0(p) + (1-\pi_0) f_1(p)
\]</span></p>
<p>where <span class="math inline">\(f(p)\)</span> is the probability density function (p.d.f.) of the observed <span class="math inline">\(p\)</span>-value distribution, <span class="math inline">\(f_0\)</span> and <span class="math inline">\(f_1\)</span> are the p.d.f. of the <span class="math inline">\(p\)</span>-value distribution under the null and alternative distribution, respectively, and <span class="math inline">\(\pi_0 = \text{Pr}(H_0)\)</span>, the prior probability of a gene being unaffected by treatment (prior as in, before observing any data), .</p>
<p>As before, we can imagine that our <span class="math inline">\(p\)</span>-values are generated by a process that first samples a hidden variable <span class="math inline">\(Z=0\)</span> with probability <span class="math inline">\(\pi_0\)</span> and <span class="math inline">\(Z=1\)</span> with probability <span class="math inline">\(1-\pi_0\)</span>, and then samples a <span class="math inline">\(p\)</span>-value (or <span class="math inline">\(t\)</span>-statistic) from the null or alternative distribution depending on the value of <span class="math inline">\(Z\)</span>. We know the null distribution is the uniform distribution. If we knew a parametric form for the alternative distribution, we could apply EM to estimate <span class="math inline">\(\pi_0\)</span> and the parameters of the alternative distribution, and then compute for each gene the <em>recognition distribution</em> or <em>local false discovery rate</em> (lfdr)</p>
<p><span class="math display">\[
\begin{aligned}
  \text{lfdr}(p)  &amp;= \text{Pr}(H_0 \mid p)
  = \text{Pr}(Z=0 \mid p)
  = \frac{\pi_0 f_0(p)}{f(p)}
\end{aligned}
\]</span></p>
<p>The word <em>“local”</em> refers to the fact that the above expression gives the expected rate of false positives among all genes with <span class="math inline">\(p\)</span>-value equal to <span class="math inline">\(p\)</span>, as opposed to the <em>“tail”</em> FDR estimated in the previous section giving the he expected rate of false positive among all genes with <span class="math inline">\(p\)</span>-value less than or equal to <span class="math inline">\(p\)</span>.</p>
<p>In reality, EM is rarely used in this context, because we rarely have a good enough idea about a parametric form for the alternative distribution. Instead a non-parametric approach is used that exploits the knowledge that:</p>
<ol type="1">
<li><p><span class="math inline">\(p\)</span>-values are uniformly distributed under the null hypothesis, that is</p>
<p><span class="math display">\[
f_0(p) = 1
\]</span></p></li>
<li><p><span class="math inline">\(p\)</span>-values close to 1 almost surely come from the null distribution, that is,</p>
<p><span class="math display">\[
  \begin{aligned}
\lim_{p\to 1} f_1(p) &amp;= 0 \\\\
\lim_{p\to 1} f(p) &amp;= \pi_0 \lim_{p\to 1} f_0(p) = \pi_0
  \end{aligned}
  \]</span></p></li>
</ol>
<p>Hence in the observed <span class="math inline">\(p\)</span>-value histogram, <span class="math inline">\(\pi_0\)</span> can be estimated from the height of the “flat” region near <span class="math inline">\(p\approx 1\)</span>.</p>
<p>We obtain</p>
<p><span class="math display">\[
\text{lfdr}(p) = \frac{\hat{\pi}_0}{f(p)}
\]</span></p>
<p>It remains to estimate the p.d.f. of the real <span class="math inline">\(p\)</span>-value distribution <span class="math inline">\(f(p)\)</span>. In the popular <a href="https://github.com/StoreyLab/qvalue">q-value package</a>, this is done by <a href="https://en.wikipedia.org/wiki/Kernel_density_estimation">kernel density estimation</a> followed by some smoothing.</p>
<p>The local false discovery rate provide a useful measure for the importance of a feature with <span class="math inline">\(p\)</span>-value <span class="math inline">\(p\)</span>. However, <span class="math inline">\(p\)</span>-values themselves represent <em>tail</em> probabilities, <span class="math inline">\(p=\text{Pr}(P\leq p \mid H_0)\)</span>, see above. Similarly, we can compute the FDR among all features with <span class="math inline">\(p\)</span>-value less than some threshold <span class="math inline">\(p\)</span>:</p>
<p><span class="math display">\[
\text{FDR}(p) = \text{Pr}(H_0\mid P\leq p).
\]</span></p>
<p>This tells us the estimated false discovery rate among all features that are equally or more significant than a feature with <span class="math inline">\(p\)</span>-value <span class="math inline">\(p\)</span>.</p>
<p>Writing <span class="math inline">\(F(p)\)</span>, <span class="math inline">\(F_0(p)\)</span> and <span class="math inline">\(F_1(p)\)</span> for the cumulative probability functions of the observed, null, and alternative <span class="math inline">\(p\)</span>-value distributions, the reasoning above can be repeated to obtain</p>
<p><span class="math display">\[
F(p)=\pi_0 F_0(p) + (1-\pi_0) F_1(p) = \pi_0 p + (1-\pi_0) F_1(p)
\]</span></p>
<p>where we used <span class="math inline">\(F_0(p)=p\)</span>. It follows that</p>
<p><span class="math display">\[
\text{FDR}(p) = \text{Pr}(H_0\mid P\leq p) = \text{Pr}(Z=0\mid P\leq p) =\frac{\pi_0 p}{F(p)}
\]</span></p>
</section>
<section id="q-values" class="level3">
<h3 class="anchored" data-anchor-id="q-values"><span class="math inline">\(Q\)</span>-values</h3>
<p>The <span class="math inline">\(q\)</span>-value of a feature with <span class="math inline">\(p\)</span>-value <span class="math inline">\(p\)</span> is defined as the minimum FDR that can be attained when calling that feature significant, that is, by choosing a <span class="math inline">\(p\)</span>-value threshold <span class="math inline">\(r\geq p\)</span>. Hence</p>
<p><span class="math display">\[
q(p) = \min_{r\geq p} \text{FDR}(r)
\]</span></p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="../statistical-significance/statistical-significance-gws.html" class="pagination-link  aria-label=" path-breaking="" paper"="">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Path-breaking paper</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../statistical-significance/statistical-significance-notebook.html" class="pagination-link" aria-label="Implementation">
        <span class="nav-page-text">Implementation</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>