<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BINF301 Machine Learning for Systems Biology - Mixture distributions</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">BINF301 Machine Learning for Systems Biology</span>
    </a>
  </div>
        <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Mixture distributions</li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../julia-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction to Julia</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Cluster analysis</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/cancer-subtypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/combinatorial-clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/cluster-analysis-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">Statistical significance</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/statistical-significance-gws.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/fdr-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/statistical-significance-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Regularized regression</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/drug-sensitivity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/ridge-lasso-elnet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/regularized-regression-notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">Dimensionality reduction</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory I</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/tsne-umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory II</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">Causal inference</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-assoc-genes-disease.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/mediation-iv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-model-selection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Implementation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Graphical models</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/gene-regulatory-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/bayesian-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Gaussian processes</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gaussian-processes/variance-components-genomics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gaussian-processes/gaussian-processes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">Neural networks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../neural-networks/protein-structure-prediction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Path-breaking paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../neural-networks/neural-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Theory</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">Appendix</span></span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#generative-models" id="toc-generative-models" class="nav-link active" data-scroll-target="#generative-models">Generative models</a></li>
  <li><a href="#gaussian-mixture-models" id="toc-gaussian-mixture-models" class="nav-link" data-scroll-target="#gaussian-mixture-models">Gaussian mixture models</a></li>
  <li><a href="#some-em-language" id="toc-some-em-language" class="nav-link" data-scroll-target="#some-em-language">Some EM language</a></li>
  <li><a href="#maximum-likelihood-estimation" id="toc-maximum-likelihood-estimation" class="nav-link" data-scroll-target="#maximum-likelihood-estimation">Maximum-likelihood estimation</a></li>
  <li><a href="#the-em-algorithm" id="toc-the-em-algorithm" class="nav-link" data-scroll-target="#the-em-algorithm">The EM algorithm</a></li>
  <li><a href="#what-are-soft-cluster-assignments" id="toc-what-are-soft-cluster-assignments" class="nav-link" data-scroll-target="#what-are-soft-cluster-assignments">What are “soft cluster assignments”?</a></li>
  <li><a href="#generalizations" id="toc-generalizations" class="nav-link" data-scroll-target="#generalizations">Generalizations</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Mixture distributions</h1>
</div>



<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Book sections
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="https://www.microsoft.com/en-us/research/publication/pattern-recognition-machine-learning/"><em>Pattern Recognition and Machine Learning</em></a>: <strong>Chapter 9</strong></li>
<li><a href="https://hastie.su.domains/ElemStatLearn/"><em>Elements of Statistical Learning</em></a>: <strong>Section 14.3</strong></li>
<li><a href="https://www.statlearning.com/"><em>An Introduction to Statistical Learning</em></a>: <strong>Section 12.4</strong></li>
</ul>
</div>
</div>
<section id="generative-models" class="level3">
<h3 class="anchored" data-anchor-id="generative-models">Generative models</h3>
<p>A generative model is a statistical model for the process that <em>could have</em> generated your data. Generative models offer many advantages compared to combinatorial algorithms that treat data as a collection of objects. Most importantly, working with a generative model forces you to be explicit about your assumptions. Likewise, a generative model allows you to encode, and be explicit about, prior (biological) knowledge you may have about the data generating process.</p>
</section>
<section id="gaussian-mixture-models" class="level3">
<h3 class="anchored" data-anchor-id="gaussian-mixture-models">Gaussian mixture models</h3>
<p>Let’s assume there is an unmeasured (or hidden) random variable <span class="math inline">\(Z\)</span> that determines to which group an observation <span class="math inline">\(X\)</span> belongs. For simplicity, assume that <span class="math inline">\(Z\)</span> can only take two values, 0 or 1, and that the measurement <span class="math inline">\(X\)</span> is one-dimensional and normally distributed in each group.</p>
<p>Consider the following process:</p>
<ol type="1">
<li><p>Randomly sample cluster label <span class="math inline">\(Z=1\)</span> with probability <span class="math inline">\(\pi\)</span> and <span class="math inline">\(Z=0\)</span> with probability <span class="math inline">\(1-\pi\)</span>.</p></li>
<li><p>Sample features</p></li>
</ol>
<p><span class="math display">\[
\begin{aligned}
X \sim \begin{cases}
\mathcal{N}(\mu_0,\sigma_0^2) &amp;  \text{ if } Z=0\\
\mathcal{N}(\mu_1,\sigma_1^2) &amp;  \text{ if } Z=1
\end{cases}
\end{aligned}
\]</span></p>
<p>Tthe model generates cluster labels <span class="math inline">\(Z\)</span> and real numbers <span class="math inline">\(x\in\mathbb{R}\)</span> from the model</p>
<p><span class="math display">\[
\begin{aligned}
Z &amp;\longrightarrow X\\\\
p(Z,x) = P(Z) &amp;\times p(x\mid Z)
\end{aligned}
\]</span></p>
<p>where we use lower-case “p” for probability density functions (<span class="math inline">\(X\)</span> is continuous) and upper-case “P” for discrete probabilities, and</p>
<p><span class="math display">\[
\begin{aligned}
P(Z=1) &amp;= \pi = 1 - P(Z=0) \\\\
p(x\mid Z=k) &amp;= \mathcal{N}(\mu_k,\sigma_k^2)
\end{aligned}
\]</span></p>
</section>
<section id="some-em-language" class="level3">
<h3 class="anchored" data-anchor-id="some-em-language">Some EM language</h3>
<p>The <strong>joint distribution</strong> is the probability distribution of a cluster label <span class="math inline">\(Z\)</span> and feature value <span class="math inline">\(x\)</span> both being produced by the model:</p>
<p><span class="math display">\[
p(Z,x) = p(x\mid Z)\\; P(Z)
\]</span></p>
<p>The <strong>marginal distribution</strong> is th probability distribution that the model produces a feature value <span class="math inline">\(x\)</span>:</p>
<p><span class="math display">\[
p(x) = \sum_{k=0,1} p(x\mid Z=k)\\; P(Z=k)
\]</span></p>
<p>The <strong>responsibility</strong> of <span class="math inline">\(Z\)</span> for feature value <span class="math inline">\(x\)</span>, also called the <strong>recognition distribution</strong>, is obtained using Bayes’ theorem</p>
<p><span class="math display">\[
P(Z=k\mid x) = \frac{p(x\mid Z=k) \\; P(Z=k)}{p(x)}
\]</span></p>
<p>This value can be used as a <strong>soft cluster assignment</strong>: with probability <span class="math inline">\(P(Z=k \mid x)\)</span>, an observed value <span class="math inline">\(x\)</span> belongs to cluster k.</p>
<p>Note that the expected value of <span class="math inline">\(Z\)</span> given a data point <span class="math inline">\(x\)</span> is:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbb{E}\left(Z\mid x\right) &amp;= 1 \cdot P(Z=1 \mid x) + 0\cdot P(Z=0 \mid x) = P(Z=1 \mid x)
% &amp;= \frac{\pi p(x\mid \mu_1, \igma_1^2)}{\pi p(x \mid \mu_1, \sigma_1^2) + (1-\pi) p(x \mid \mu_0, \sigma_0^2)}
\end{aligned}
\]</span></p>
</section>
<section id="maximum-likelihood-estimation" class="level3">
<h3 class="anchored" data-anchor-id="maximum-likelihood-estimation">Maximum-likelihood estimation</h3>
<p>To fit the model to the data, we can only use the observed data <span class="math inline">\(x\)</span>, which follows the <strong>Gaussian mixture distribution</strong></p>
<p><span class="math display">\[
p(x) = \sum_{k=0,1} p(x\mid Z=k)\\; P(Z=k)
\]</span></p>
<p>The log-likelihood of observing <span class="math inline">\(N\)</span> independent samples <span class="math inline">\((x_1,\dots,x_N)\)</span> is</p>
<p><span class="math display">\[
\mathcal{L}= \log\left(\prod_{i=1}^N p(x_{i}) \right) = \sum_{i=1}^N \log p(x_{i})
\]</span></p>
<p>We want to find the best-fitting model by maximizing the log-likelihood.</p>
<p>Directly maximizing the log-likelihood with respect to the paramaters <span class="math inline">\(\pi\)</span>, <span class="math inline">\(\mu_k\)</span>, and <span class="math inline">\(\sigma_k^2\)</span> is difficult, because:</p>
<ul>
<li>Only the feature values <span class="math inline">\(x\)</span> are observed.</li>
<li>The cluster labels <span class="math inline">\(Z\)</span> are hidden, they are latent variables.</li>
<li>The log-likelihood is expressed purely in terms of the observable distribution, involves logarithms of sums, and is intractable.</li>
</ul>
<p>If we knew the cluster labels <span class="math inline">\(k\)</span> for each sample, we could easily fit the parameters <span class="math inline">\((\pi,\mu_k,\sigma_k^2)\)</span> from the data for each cluster.</p>
<p>If we knew the parameters <span class="math inline">\((\pi, \mu_k,\sigma_k^2)\)</span>, we could easily determine the probability for each data point to belong to each cluster and determine cluster labels.</p>
<p>To get around this catch-22, we replace actual cluster labels by their current expected values given current values for the parameters, and then iterate the above two steps until convergence - this is the Expectation-Maximization (EM) algorithm.</p>
</section>
<section id="the-em-algorithm" class="level3">
<h3 class="anchored" data-anchor-id="the-em-algorithm">The EM algorithm</h3>
<ol type="1">
<li>Take initial guesses <span class="math inline">\(\hat\pi\)</span>, <span class="math inline">\(\hat\mu_k\)</span>, <span class="math inline">\(\hat\sigma_k^2\)</span> for the model parameters.</li>
<li>Expectation step: Compute the responsibilities <span class="math inline">\(P(Z_i=k\mid x_i)\)</span> for each data point <span class="math inline">\(x_i\)</span>.</li>
<li>Maximization step: Update <span class="math inline">\(\hat\pi\)</span>, <span class="math inline">\(\hat\mu_k\)</span>, <span class="math inline">\(\hat\sigma_k^2\)</span> by maximizing the log-likelihood using the soft cluster assignments <span class="math inline">\(P(Z_i=k\mid x_i)\)</span>.</li>
<li>Iterate steps 2 and 3 until convergence.</li>
</ol>
</section>
<section id="what-are-soft-cluster-assignments" class="level3">
<h3 class="anchored" data-anchor-id="what-are-soft-cluster-assignments">What are “soft cluster assignments”?</h3>
<p>Consider <span class="math inline">\(N\)</span> samples <span class="math inline">\((x_1,\dots,x_N)\)</span> from a normal distribution <span class="math inline">\(p(x\mid \mu,\sigma^2)\)</span>. The log-likelihood is</p>
<p><span class="math display">\[
\begin{aligned}
\mathcal{L}= \sum_{i=1}^N \log \left(\frac1{\sqrt{2\pi\sigma^2}} e^{-\frac{1}{2\sigma^2}(x_i-\mu)^2}\right) = -\frac{N}{2} \log(\sigma^2) - \frac{1}{2\sigma^2} \sum_{i=1}^N (x_i-\mu)^2
\end{aligned}
\]</span></p>
<p><span class="math inline">\(\mathcal{L}\)</span> is maximized for</p>
<p><span class="math display">\[
\begin{aligned}
\hat\mu &amp;= \frac{1}{N} \sum_{i=1}^N x_i\\\\
\hat\sigma^2 &amp;= \frac{1}{N} \sum_{i=1}^N (x_i-\hat\mu)^2
\end{aligned}
\]</span></p>
<p>Now consider <span class="math inline">\(N\)</span> samples <span class="math inline">\((Z_1,\dots,Z_N)\)</span> and <span class="math inline">\((x_1,\dots,x_N)\)</span> from the generative model where the cluster labels are also observed. The log-likelihood is</p>
<p><span class="math display">\[
\begin{aligned}
\mathcal{L}&amp;=  \sum_{i=1}^N \log p(Z_i,x_i)\\\\
&amp;= \sum_{i=1}^N \Bigl(Z_i \log p(Z_i=1,x_i) + (1-Z_i) \log p(Z_i=0,x_i)\Bigr) \\\\
&amp;= \sum_{i=1}^N \Bigl(Z_i \log p(x_i\mid \mu_1,\sigma_1^2) + (1-Z_i) \log p(x_i\mid \mu_0,\sigma_0^2)\Bigr) + \sum_{i=1}^N \Bigl(Z_i\log\pi + (1-Z_i) \log(1-\pi)\Bigr)
\end{aligned}
\]</span></p>
<p><span class="math inline">\(\mathcal{L}\)</span> is maximized for</p>
<p><span class="math display">\[
\begin{aligned}
\hat\pi &amp;= \frac{N_1}{N}\\\\
\hat\mu_k &amp;= \frac{1}{N_k} \sum_{Z_i=k} x_i\\\\
\hat\sigma_k^2 &amp;= \frac{1}{N_k} \sum_{Z_i=k}^N (x_i-\hat\mu_k)^2
\end{aligned}
\]</span></p>
<p>Since the cluster labels are not observed, we don’t know the value of the <span class="math inline">\(Z_i\)</span>. The “trick” is to replace them with their expectation values <span class="math inline">\(\mathbb{E}(Z_i=k) = P(Z_i=k\mid x_i)\)</span> in the EM algorithm, using the current estimates for <span class="math inline">\(\hat\pi\)</span>, <span class="math inline">\(\hat\mu_k\)</span>, <span class="math inline">\(\hat\sigma_k^2\)</span>.</p>
<p>This leads to updated estimates</p>
<p><span class="math display">\[
\begin{aligned}
\hat\pi^{\text{(new)}} &amp;= \frac{\sum_{i=1}^N P(Z_i=k\mid x_i)}{N}\\\\
\hat\mu_k^{\text{(new)}} &amp;= \frac{1}{N} \sum_{i=1}^N P(Z_i=k\mid x_i)\\;  x_i \\\\
(\hat\sigma_k^2)^{\text{(new)}} &amp;= \frac{1}{N} \sum_{i=1}^N P(Z_i=k\mid x_i)\\; (x_i-\hat\mu_k)^2
\end{aligned}
\]</span></p>
<p>Hence, instead of a “hard” assignment of each sample <span class="math inline">\(x_i\)</span> to one cluster <span class="math inline">\(k\)</span> when the <span class="math inline">\(Z_i\)</span> are observed, each sample now contributes with a “soft assignment” weight <span class="math inline">\(P(Z_i=k\mid x_i)\)</span> to the parameters of each cluster.</p>
<p>After convergence, the final <span class="math inline">\(P(Z_i=k\mid x_i)\)</span> can be used to assign data points to clusters, for instance to the cluster <span class="math inline">\(k\)</span> with highest responsibility for <span class="math inline">\(x_i\)</span>.</p>
</section>
<section id="generalizations" class="level3">
<h3 class="anchored" data-anchor-id="generalizations">Generalizations</h3>
<p>We have so far considered the case where the data are one-dimensional (real numbers) and the number of clusters is pre-fixed. Important generalizations are:</p>
<ol type="1">
<li><p>The data can be of any dimension <span class="math inline">\(D\)</span>. In higher dimensions the components of the mixture model are <em>multivariate</em> normal distributions. The mean parameters <span class="math inline">\(\mu_k\)</span> simply become <span class="math inline">\(D\)</span>-dimensional vectors. The variance parameters <span class="math inline">\(\sigma_k^2\)</span> however become <span class="math inline">\(D\times D\)</span> covariance <em>matrices</em>. For simplicity and to reduce the number of paramaters, it is often assumed that the covariance matrices are diagonal, such that the number of variance parameters again scales linearly in <span class="math inline">\(D\)</span>. However, when features are correlated, this assumption will be a poor representation of the underlying data.</p></li>
<li><p>Instead of treating the cluster weights, means, and variances as fixed parameters that need to be estimated, they can be seen as random variables themselves with a distribution (uncertainty) that can be learned from the data. For more information, see this <a href="https://turinglang.org/v0.24/tutorials/01-gaussian-mixture-model/">tutorial</a>.</p></li>
<li><p>In an <em>infinite mixture model</em>, the number of clusters need not be fixed in advance, but can be learned from the data. For more information, see this <a href="https://turinglang.org/v0.24/tutorials/06-infinite-mixture-model/">tutorial</a>.</p></li>
</ol>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>