---
title: "Path-breaking paper: Unbiased classification of sensory neuron types by large-scale single-cell RNA sequencing"
page-layout: full
title-block-banner: false
---

::: {.callout-tip}
## Path-breaking paper

Usoskin D et al. [*Unbiased classification of sensory neuron types by large-scale single-cell RNA sequencing*](https://doi.org/10.1038/nn.3881) . Nature Neuroscience **18**:145 (2015).
:::

::: {.callout-note}
## Test of time paper

Tasic B et al. [*Shared and distinct transcriptomic cell types across neocortical areas*](https://doi.org/10.1038/s41586-018-0654-5). Nature 563:72 (2018).

Siletti K et al. [*Transcriptomic diversity of cell types across the adult human brain*](https://doi.org/10.1126/science.add7046). Science 382:eadd7046 (2023). See also the [Brain Cell Census collection](https://www.science.org/collections/brain-cell-census)
:::

### Motivation

Reducing high-dimensional data to a low-dimensional representation for the purposes of visualization and pattern discovery is a classical approach in statistics, with [principal component analysis (PCA)](https://en.wikipedia.org/wiki/Principal_component_analysis) probably the best known method. PCA and other [dimensionality reduction](https://en.wikipedia.org/wiki/Dimensionality_reduction) methods have been used in systems biology for as long as (relatively) large data have been around. Nowadays, [dimensionality reduction](https://en.wikipedia.org/wiki/Dimensionality_reduction) is mostly associated with the analysis of [single-cell sequencing](https://en.wikipedia.org/wiki/Single-cell_sequencing) data. Given the importance of [single-cell sequencing](https://en.wikipedia.org/wiki/Single-cell_sequencing) technologies, and the relative lack of understanding of the [non-linear dimensionality reduction](https://en.wikipedia.org/wiki/Nonlinear_dimensionality_reduction) methods that are regularly employed in this context, this will be our angle for learning about [dimensionality reduction](https://en.wikipedia.org/wiki/Dimensionality_reduction).

For our path-breaking paper, we select a relatively early study of cell types in the brain using [single-cell sequencing](https://en.wikipedia.org/wiki/Single-cell_sequencing) of 622 mouse neurons. In this study, PCA is used to visualize data in two and three dimensions and to identify distinct types of sensory neurons.

To illustrate how far the field of [single-cell sequencing](https://en.wikipedia.org/wiki/Single-cell_sequencing) has advanced in the last decade, we select a paper from 2018, which sequenced more than 23 thousand cells and whose data we will use in our practical analysis, and a paper from the [BRAIN Initiative Cell Census Network](https://braininitiative.nih.gov/research/tools-and-technologies-brain-cells-and-circuits/brain-initiative-cell-census-network), which sequenced more than three million human brain cells, including more than two million neurons. In these papers, [t-SNE](https://en.wikipedia.org/wiki/T-distributed_stochastic_neighbor_embedding) and [UMAP](https://en.wikipedia.org/wiki/Nonlinear_dimensionality_reduction#Uniform_manifold_approximation_and_projection) are used for dimensionality reduction, as is the case in most current studies in the field.

### Questions for discussion

What is single-cell transcriptomics? We will look in more detail at this review paper: [*Revealing the vectors of cellular identity with Single-cell genomics.*](https://www.nature.com/articles/nbt.3711)

What is the basic idea for discovering and classifying cell types?

How many expressed genes were detected on average in each cell? How many genes were detected in total across all cells (see Methods)? What could explain these numbers?

How did the authors decide on the initial classification of genes in neuronal and non-neuronal clusters? How did they define further subclusters of the initial neuronal clusters?

What is shown in Figure 1?

What is shown in Figure 4? How is the "fraction of positive cells" defined for a given gene?