<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BINF301 Genome-scale Algorithms - Variance component models and Gaussian processes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">BINF301 Genome-scale Algorithms</span>
    </a>
  </div>
        <div class="quarto-navbar-tools">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../spatio-temporal-models/spatial-gene-expression.html">Spatio-Temporal models</a></li><li class="breadcrumb-item"><a href="../spatio-temporal-models/gaussian-processes.html">Variance component models and Gaussian processes</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../julia-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction to Julia</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Cluster analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/cancer-subtypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cancer subtypes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/combinatorial-clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Combinatorial clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../cluster-analysis/mixture-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mixture distributions</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Statistical significance</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/statistical-significance-gws.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Statistical significance for genome-wide studies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../statistical-significance/fdr-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">False discovery rate estimation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Regularized regression</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/drug-sensitivity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Drug sensitivity prediction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../regularized-regression/ridge-lasso-elnet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Ridge, lasso and elastic net regression</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Dimensionality reduction</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PCA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../dimensionality-reduction/tsne-umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">t-SNE, UMAP</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Causal inference</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-assoc-genes-disease.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Causal associations between gene expression and disease</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/mediation-iv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mediation and instrumental variable analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal-inference/causal-model-selection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Causal model selection</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true">
 <span class="menu-text">Graphical models</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/gene-regulatory-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene regulatory networks</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphical-models/bayesian-networks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bayesian networks</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true">
 <span class="menu-text">Spatio-Temporal models</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../spatio-temporal-models/spatial-gene-expression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial gene experssion</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../spatio-temporal-models/gaussian-processes.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Variance component models and Gaussian processes</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">Appendix</span></span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#linear-regression-from-fixed-to-random-coefficients" id="toc-linear-regression-from-fixed-to-random-coefficients" class="nav-link active" data-scroll-target="#linear-regression-from-fixed-to-random-coefficients">Linear regression, from fixed to random coefficients</a></li>
  <li><a href="#kernel-based-variance-component-models" id="toc-kernel-based-variance-component-models" class="nav-link" data-scroll-target="#kernel-based-variance-component-models">Kernel-based variance component models</a></li>
  <li><a href="#gaussian-processes" id="toc-gaussian-processes" class="nav-link" data-scroll-target="#gaussian-processes">Gaussian processes</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../spatio-temporal-models/spatial-gene-expression.html">Spatio-Temporal models</a></li><li class="breadcrumb-item"><a href="../spatio-temporal-models/gaussian-processes.html">Variance component models and Gaussian processes</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Variance component models and Gaussian processes</h1>
</div>



<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Book sections
</div>
</div>
<div class="callout-body-container callout-body">
<p><a href="https://www.microsoft.com/en-us/research/publication/pattern-recognition-machine-learning/"><em>Pattern Recognition and Machine Learning</em></a>: <strong>Chapter 6</strong></p>
</div>
</div>
<section id="linear-regression-from-fixed-to-random-coefficients" class="level3">
<h3 class="anchored" data-anchor-id="linear-regression-from-fixed-to-random-coefficients">Linear regression, from fixed to random coefficients</h3>
<p>In a standard linear regression model for an outcome <span class="math inline">\(Y\)</span> on one or more predictors <span class="math inline">\(X_1,\dots,X_p\)</span></p>
<p><span class="math display">\[
Y = \sum_i\beta_i X_i + \epsilon = X^T\beta + \epsilon, \qquad
  \epsilon \sim \mathcal{N}(0,\sigma_e^2)
\]</span></p>
<p>the effects <span class="math inline">\(\beta_i\)</span> are treated as <strong>fixed</strong>, such that the distribution of <span class="math inline">\(Y\)</span> is:</p>
<p><span class="math display">\[
p(Y\mid X) = \mathcal{N}(X^T\beta,\sigma_e^2),
\]</span></p>
<p>that is, in a <strong>fixed effect model</strong>, the predictors affect the average or expected value of <span class="math inline">\(Y\)</span>. In a fixed effect model, we always assume that the data to fit the parameters are obtained from <strong>independent samples</strong> of an underlying model.</p>
<p>If our data consists of <strong>non-independent samples</strong> (e.g.&nbsp;due to temporal, spatial or familial relations among the observations), we can use a <strong>random effect model</strong> to explain these correlations as a function of covariates <span class="math inline">\(X\)</span>. As <a href="../../regularized-regression/ridge-lasso-elnet/">before</a>, we collect the data for the inputs and ouptput in a <span class="math inline">\(N\times p\)</span> matrix <span class="math inline">\(\mathbf{X}\)</span> and <span class="math inline">\(N\)</span>-vector <span class="math inline">\(\mathbf{y}\)</span>, respectively, <span class="math display">\[\begin{aligned}
    \mathbf{X}= (x_1,x_2,\dots,x_p) = \begin{pmatrix}
      x_{11} &amp; x_{12} &amp; \dots &amp; x_{1p}\\
      x_{21} &amp; x_{22} &amp; \dots &amp; x_{2p}\\
      \vdots &amp; \vdots &amp; &amp; \vdots\\
      x_{N1} &amp; x_{N2} &amp; \dots &amp; x_{Np}
    \end{pmatrix} &amp;&amp;
    \mathbf{y}= \begin{pmatrix}
      y_1\\
      y_2\\
      \vdots\\
      y_N
    \end{pmatrix}
  \end{aligned}\]</span></p>
<p>We model the data as</p>
<p><span class="math display">\[
\begin{aligned}
  \mathbf{y} &amp;= \mathbf{X} \alpha + \epsilon\
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\alpha = (\alpha_1,\dots,\alpha_p)^T\)</span> are random effects with multi-variate normal distribution</p>
<p><span class="math display">\[
\alpha \sim \mathcal{N}(0,\sigma_a^2 I_p)
\]</span></p>
<p>and <span class="math inline">\(\epsilon = (\epsilon_1,\dots,\epsilon_N)\)</span> are independent error terms</p>
<p><span class="math display">\[
\epsilon \sim \mathcal{N}(0,\sigma_e^2 I_N).
\]</span></p>
<p>The random effects are assumed to be independent or the errors. We assumed the random effects are mutually independent, but this can easily be generalized.</p>
<p>Using <a href="https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Affine_transformation">properties of the multivariate normal distribution</a>, the distribution of <span class="math inline">\(\mathbf{y}\)</span> can be seen to be:</p>
<p><span class="math display">\[
p(\mathbf{y}\mid \mathbf{X}) = \mathcal{N}(0, \sigma_a^2 \mathbf{X}\mathbf{X}^T + \sigma_e^2 I_N)
\]</span></p>
<p>and we see that <span class="math inline">\(\mathbf{X}\)</span>, or more precisely the covariance matrix <span class="math inline">\(\mathbf{X}\mathbf{X}^T\)</span> indeed models correlations among the samples in <span class="math inline">\(\mathbf{y}\)</span>.</p>
<p>Naturally, fixed and random effects can be combined into a so-called <a href="https://en.wikipedia.org/wiki/Mixed_model">mixed model</a>.</p>
<p>Models of this kind are often used in genetics, where the random effect covariates <span class="math inline">\(\mathbf{X}\)</span> are genetic markers and their covariance matrix <span class="math inline">\(\mathbf{X}\mathbf{X}^T\)</span> expresses the genetic similarity between individuals in the study.</p>
<p>So far we assumed that all random effects had the same variance <span class="math inline">\(\sigma_a^2\)</span>. If we assume that each <span class="math inline">\(\alpha_j\)</span> has a different variance <span class="math inline">\(\sigma_j^2\)</span>, we would obtain</p>
<p><span class="math display">\[
p(\mathbf{y}\mid \mathbf{X}) = \mathcal{N}(0, \mathbf{K})
\]</span></p>
<p>where</p>
<p><span class="math display">\[
\mathbf{K} = \sum_{j=1}^p \sigma_j^2 \mathbf{x}_j \mathbf{x}_j^T  + \sigma_e^2 I_N
\]</span></p>
<p>that is, <span class="math inline">\(\sigma_j^2\)</span> measures the contribution of covariate <span class="math inline">\(X_j\)</span> to the correlations among samples of <span class="math inline">\(Y\)</span>.</p>
<p>To estimate the variance parameters, maximum-likelihood is employed, that is, we find the values of the <span class="math inline">\(\sigma_j^2\)</span> and <span class="math inline">\(\sigma_e^2\)</span> which maximize</p>
<p><span class="math display">\[
\log p(\mathbf{y}) = -\frac12 \log \det (\mathbf{K} ) - \frac12 \mathbf{y}^T \mathbf{K}^{-1} \mathbf{y} - \frac{N}2 \log(2\pi)
\]</span></p>
<p>This is a difficult, non-convex optimization problem which requires the use of numerical, gradient-based optimizers.</p>
<p>Note that the covariance matrix satisfies</p>
<p><span class="math display">\[
\begin{aligned}
  \mathbb{E}(y_i y_j) = K_{ij}
\end{aligned}
\]</span> and hence the total variance of <span class="math inline">\(\mathbf{y}\)</span> can be written as</p>
<p><span class="math display">\[
\begin{aligned}
  \mathbb{E}(\mathbf{y}^T\mathbf{y}) = \sum_i \mathbb{E}(y_i^2) = \sum_i K_{ii} = \mathrm{tr}(K)
\end{aligned}
\]</span></p>
<p>From the definition of <span class="math inline">\(\mathbf{K}\)</span>, its trace can be written as</p>
<p><span class="math display">\[
\mathrm{tr}(K) = \sum_{j=1}^p \sigma_j^2 \mathbf{x}_j^T \mathbf{x}_j   + N \sigma_e^2.
\]</span></p>
<p>Therefore we say that</p>
<p><span class="math display">\[
\frac{\sigma_j^2 \mathbf{x}_j^T \mathbf{x}_j}{\mathrm{tr}(K) }
\]</span></p>
<p>is the <strong>variance in <span class="math inline">\(\mathbf{y}\)</span> explained by variance component <span class="math inline">\(\mathbf{x}_i\)</span></strong>.</p>
</section>
<section id="kernel-based-variance-component-models" class="level3">
<h3 class="anchored" data-anchor-id="kernel-based-variance-component-models">Kernel-based variance component models</h3>
<p>Since the posterior distribution <span class="math inline">\(p(\mathbf{y}\mid \mathbf{X})\)</span> only depends on the matrix <span class="math inline">\(\mathbf{K}\)</span>, an immediate generalization is to define variance component models directly in terms of <strong>kernel matrices</strong> instead of starting from a linear, random effect model, namely define</p>
<p><span class="math display">\[
p(\mathbf{y}) = \mathcal{N}(0, \mathbf{K})
\]</span></p>
<p>For instance, if the elements (samples) <span class="math inline">\(y_i\)</span> of <span class="math inline">\(\mathbf{y}\)</span> are obtained at specific locations (e.g.&nbsp;pixels in an image) <span class="math inline">\(\mathbf{x_i}\in \mathbb{R}^2\)</span> (or <span class="math inline">\(\mathbb{R}^p\)</span> more generally), we can define</p>
<p><span class="math display">\[
K_{ij} = k(\mathbf{x}_i,\mathbf{x}_j)
\]</span></p>
<p>where the <strong>kernel function</strong> <span class="math inline">\(k\)</span> is a function that measures the similarity or closeness between samples.</p>
<p>Generalizing from before, the kernel matrix <span class="math inline">\(\mathbf{K}\)</span> can consist of multiple components,</p>
<p><span class="math display">\[
\mathbf{K} = \sum_{j=1}^p \sigma_j^2 \mathbf{K}_j  + \sigma_e^2 I_N
\]</span></p>
<p>and the <strong>variance explained</strong> by the <span class="math inline">\(j^{\text{th}}\)</span> component is</p>
<p><span class="math display">\[
\frac{\sigma_j^2\mathrm{tr}(\mathbf{K}_j)}{\mathrm{tr}(\mathbf{K})}
\]</span></p>
</section>
<section id="gaussian-processes" class="level3">
<h3 class="anchored" data-anchor-id="gaussian-processes">Gaussian processes</h3>
<p>In many applications, the “positions” <span class="math inline">\(\mathbf{x}_i\)</span> where samples are obtained are not fixed, but a finite subset of a possibly infinite, continuous range. For instance, when studying a dynamic process, we typically obtain noisy measurements <span class="math inline">\(y_i\)</span> at a finite number of time points <span class="math inline">\(t_i\)</span>, and are interested in the entire underlying process <span class="math inline">\(y(t)\)</span> for all times <span class="math inline">\(t\)</span> in some time interval. Similarly, we may have measurements <span class="math inline">\(y_i\)</span> at a finite number of locations <span class="math inline">\(\mathbf{x}_i\)</span>, and are interested in the entire function <span class="math inline">\(y(\mathbf{x})\)</span> for all positions <span class="math inline">\(\mathbf{x}\)</span> in some spatial region.</p>
<p>A <a href="https://en.wikipedia.org/wiki/Gaussian_process">Gaussian process</a> is a <a href="https://en.wikipedia.org/wiki/Stochastic_process">stochastic process</a>, to be understood as a probability distribution over functions <span class="math inline">\(y(\mathbf{x})\)</span> over some continuous domain <span class="math inline">\(D\)</span>, such that the set of values of <span class="math inline">\(y(\mathbf{x})\)</span> evaluated at a finite set of points <span class="math inline">\(\mathbf{x}_1,\dots,\mathbf{x}_N\in D\)</span> are jointly normally distributed. A Gaussian process is specified by a kernel function <span class="math inline">\(k(\mathbf{x},\mathbf{x}')\)</span>, for <span class="math inline">\(\mathbf{x},\mathbf{x}' \in D\)</span>. Writing <span class="math inline">\(\mathbf{y} = (y(\mathbf{x}_1), \dots, y(\mathbf{x}_N))\)</span>, the kernel defines the probability distribution</p>
<p><span class="math display">\[
p(\mathbf{y}) = \mathcal{N}(0, \mathbf{K})
\]</span></p>
<p>where</p>
<p><span class="math display">\[
K_{ij} = k(\mathbf{x}_i,\mathbf{x}_j)
\]</span></p>
<p>Hence, for the purposes of parameter estimation, the Gaussian process and variance component viewpoints are identical, which is why the term are often used interchangeably. The main difference lies in the interpretation, where Gaussian process see the data as a finite set of samples from a continuous space. Hence in the Gaussian process viewpoint, we would be particularly interested in <strong>interpolation</strong>, predicting expected values of the function <span class="math inline">\(y(\mathbf{x})\)</span> at locations <span class="math inline">\(\mathbf{x}\)</span> that were not part of the initial (training) data.</p>
<p>Keeping the notation <span class="math inline">\(\mathbf{y}\)</span> for the function values at the training positions <span class="math inline">\(\mathbf{x}_i\)</span>, we know that the joint distribution of <span class="math inline">\(\mathbf{y}\)</span> and the value of <span class="math inline">\(y(\mathbf{x})\)</span> at a new position <span class="math inline">\(\mathbf{x}\)</span> is given by</p>
<p><span class="math display">\[
p\left(
\begin{bmatrix}
  \mathbf{y}\\
  y(\mathbf{x})
\end{bmatrix}
\right) =  \mathcal{N}(0, \mathbf{K}')
\]</span></p>
<p>where <span class="math inline">\(\mathbf{K}'\)</span> is the <span class="math inline">\((N+1)\times (N+1)\)</span> matrix</p>
<p><span class="math display">\[
\mathbf{K}' =
\begin{bmatrix}
  \mathbf{K} &amp; \mathbf{k}\\
  \mathbf{k}^T &amp; c
\end{bmatrix}
\]</span></p>
<p>where</p>
<p><span class="math display">\[
\begin{aligned}
  \mathbf{K} &amp;= \left[ k(\mathbf{x}_i,\mathbf{x}_j) \right] \in \mathbb{R}^{N\times N}\\
  \mathbf{k} &amp;= k(\mathbf{x}_i,\mathbf{x}) \in \mathbb{R}^{N} \\
  c &amp;= k(\mathbf{x},\mathbf{x})\in \mathbb{R}
\end{aligned}
\]</span></p>
<p>Using <a href="https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Conditional_distributions">properties of the multivariate normal distribution</a>, the conditional distribution of the unseen value given the training data is:</p>
<p><span class="math display">\[
\begin{aligned}
  p\left( y(\mathbf{x}) \mid \mathbf{y} \right) &amp;= \mathcal{N}(\mu,\sigma^2)\\
  \mu &amp;= \mathbf{k}^T \mathbf{K}^{-1}\mathbf{y}\\
  \sigma^2 &amp;= c - \mathbf{k}^T \mathbf{K}^{-1} \mathbf{k}
\end{aligned}
\]</span></p>
<p>Generalization to interpolation to multiple points is immediate.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>